### PR TITLE
Components accessibility updates & dom optimization checks

### PR DESCRIPTION
**Pull request by aluarius**
Related to:
#1374 
#1810 

## Description
`VaColorIndicator`, `VaColorInput`, `VaDateInput`, `VaDatePicker`, `VaFileUpload`, `VaDateTable`, `VaCounter`, `VaColorPalette`  accessibility updates & dom optimization checks. Minor updates to related components.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-chip/VaChip.vue [hunk 1]
```diff
@@ -34,6 +34,10 @@
         v-if="closeable"
         class="va-chip__close-icon"
         name="close"
+        role="button"
+        aria-label="close"
+        aria-hidden="false"
+        :tabindex="indexComputed"
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-chip/VaChip.vue:40]: Will be great if you name all tab indexes as you already named below (`tabIndexComputed`).

File: packages/ui/src/components/va-counter/VaCounter.vue [hunk 2]
```diff
@@ -205,12 +210,16 @@ export default defineComponent({
         : Number(valueComputed.value) >= props.max
     })
 
+    const isDisabled = computed(() => props.readonly || props.disabled)
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-counter/VaCounter.vue:213]: They're not the same.

**Review by rustem-nasyrov — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/va-checkbox/VaCheckbox.vue [hunk 3]
```diff
@@ -90,6 +91,7 @@ export default defineComponent({
     indeterminateIcon: { type: String as PropType<string>, default: 'remove' },
     id: { type: String as PropType<string>, default: '' },
     name: { type: String as PropType<string>, default: '' },
+    ariaLabel: { type: String, default: null },
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-checkbox/VaCheckbox.vue:94]: `default: null` is it on purpose?

**Review by Derranion — APPROVED** [review]

File: packages/ui/src/components/va-input/VaInput.vue [hunk 4]
```diff
@@ -208,7 +209,7 @@ export default defineComponent({
     }
 
     const computedChildAttributes = computed(() => ({
-      ariaLabel: props.label,
+      ariaLabel: props.label || props.ariaLabel,
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-input/VaInput.vue:212]:
```suggestion
      ariaLabel: props.ariaLabel || props.label,
```

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/va-counter/VaCounter.vue [hunk 5]
```diff
@@ -205,12 +210,16 @@ export default defineComponent({
...
+
+    const tabIndexComputed = computed(() => isDisabled.value ? -1 : 0)
```
[the remark this answers is not in the document]
Comment (aluarius) [reply packages/ui/src/components/va-counter/VaCounter.vue:215]: Only interactive elements should be accessible with `Tab`.

File: packages/ui/src/components/va-date-input/VaDateInput.vue [hunk 6]
```diff
@@ -36,18 +37,27 @@
...
                 v-if="canBeCleared"
+                aria-hiden="false"
+                role="button"
+                aria-label="reset"
+                tabindex="0"
+                class="va-date-input__clear-icon"
                 v-bind="clearIconProps"
                 @click.stop="reset()"
+                @keydown.enter.stop="reset()"
+                @keydown.space.stop="reset()"
```
[the remark this answers is not in the document]
Comment (aluarius) [reply packages/ui/src/components/va-date-input/VaDateInput.vue:55]: It's related to `click.stop`, have made the change to local `click.stop`.

File: packages/ui/src/components/va-checkbox/VaCheckbox.vue (hunk 3 above)
Comment (aluarius) [reply packages/ui/src/components/va-checkbox/VaCheckbox.vue:94]: Thought it was, but after check have found out, that it's not.

File: packages/ui/src/components/va-date-input/VaDateInput.vue (hunk 6 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-date-input/VaDateInput.vue:55]: Oh sure! It's my fault :)

**Review by RVitaly1978 — APPROVED** [review]

File: packages/ui/src/components/va-counter/VaCounter.vue [hunk 7]
```diff
@@ -205,6 +208,8 @@ export default defineComponent({
         : Number(valueComputed.value) >= props.max
     })
 
+    const tabIndexComputed = computed(() => props.readonly || props.disabled ? -1 : 0)
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-counter/VaCounter.vue:211]:
```suggestion
    const tabIndexComputed = computed(() => props.disabled ? -1 : 0)
```

**Review by RVitaly1978 — APPROVED** [review]

**Review by RVitaly1978 — APPROVED** [review]

**Review by rustem-nasyrov — APPROVED** [review]

**Pull request merged**